src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -52,6 +52,13 @@ public StorageOperationException(String message) {
         }
     }
 
+    /**
+     * Signals that error has occurred while trying to write to file as file has read-only permission set
+     */
+    public static class StorageReadOnlyException extends StorageOperationException {
+        public StorageReadOnlyException(String message) { super(message); }
```

Comment:
Please check the coding standards on how to write methods with just one line of code.